Python/pystate.c

```diff
@@ -2070,20 +2070,18 @@ _PyThreadState_Attach(PyThreadState *tstate)
         }
 
 #ifdef Py_GIL_DISABLED
-        if (_PyEval_IsGILEnabled(tstate) != acquired_gil) {
+        if (_PyEval_IsGILEnabled(tstate) != tstate->_status.holds_gil) {
```

Comment:
I think this makes a bit more sense as:

```suggestion
        if (_PyEval_IsGILEnabled(tstate) && !tstate->_status.holds_gil) {
```

Although, then the following assertion is trivially true.

Reply:
I'll just kill the assertion; I agree that your suggestion is easier to follow.